filter.go

```diff
@@ -0,0 +1,95 @@
+package main
+
+import (
+	"encoding/base64"
+	"strings"
+
+	"github.com/envoyproxy/envoy/contrib/golang/filters/http/source/go/pkg/api"
+)
+
+type filter struct {
+	callbacks api.FilterCallbackHandler
+	config    *config
+}
+
+const secretKey = "secret"
+
+// parseBasicAuth parses an HTTP Basic Authentication string.
+// "Basic QWxhZGRpbjpvcGVuIHNlc2FtZQ==" returns ("Aladdin", "open sesame", true).
+func parseBasicAuth(auth string) (username, password string, ok bool) {
+	const prefix = "Basic "
+	// Case insensitive prefix match. See https://github.com/golang/go/issues/22736.
+	if len(auth) < len(prefix) || !strings.EqualFold(auth[:len(prefix)], prefix) {
+		return "", "", false
+	}
+	c, err := base64.StdEncoding.DecodeString(auth[len(prefix):])
+	if err != nil {
+		return "", "", false
+	}
+	cs := string(c)
+	username, password, ok = strings.Cut(cs, ":")
+	if !ok {
+		return "", "", false
+	}
+	return username, password, true
+}
+
+func (f *filter) verify(header api.RequestHeaderMap) (bool, string) {
+	auth, ok := header.Get("authorization")
+	if !ok {
+		return false, "no Authorization"
+	}
+	username, password, ok := parseBasicAuth(auth)
+	if !ok {
+		return false, "invalid Authorization format"
+	}
+
+	users := f.config.users
+	for _, user := range users {
```

Comment:
Would it be faster if we can convert the users to a Map?

Reply:
Resolved.